File: Backend/user-service/spring_boot_backend_template/spring_boot_backend_template/src/main/java/com/backend/entity/DroppingPoint.java
package com.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalTime;

import com.fasterxml.jackson.annotation.JsonBackReference;

@Entity
@Table(name = "dropping_points")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DroppingPoint {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String locationName;   // Andheri, Dadar, Borivali

    @Column(nullable = false)
    private LocalTime droppingTime;

    @ManyToOne
    @JsonBackReference
    @JoinColumn(name = "schedule_id", nullable = false)
    private BusSchedule busSchedule;
}

File: Backend/admin-service/admin/admin-service/Models/DroppingPoint.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace admin_service.Models
{
    [Table("dropping_points")]
    public class DroppingPoint
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Required]
        [Column("location_name")]
        public string LocationName { get; set; }  // Andheri, Dadar, Borivali

        [Required]
        [Column("dropping_time")]
        public TimeSpan DroppingTime { get; set; }

        [Required]
        [Column("schedule_id")]
        public long BusScheduleId { get; set; }

        [ForeignKey("BusScheduleId")]
        public BusSchedule BusSchedule { get; set; }
    }
}
